server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/IncludeExclude.java

```diff
@@ -395,18 +390,21 @@ public IncludeExclude(int partition, int numPartitions) {
      */
     public IncludeExclude(StreamInput in) throws IOException {
         if (in.readBoolean()) {
-            includeValues = null;
-            excludeValues = null;
-            incZeroBasedPartition = 0;
-            incNumPartitions = 0;
             String includeString = in.readOptionalString();
             include = includeString == null ? null : new RegExp(includeString);
             String excludeString = in.readOptionalString();
             exclude = excludeString == null ? null : new RegExp(excludeString);
-            return;
+            if (in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
I think this version string is fine for now.  When we backport it, we'll update it accordingly.  There's a process for doing that, but it's on the person merging the backport to do it, so you shouldn't have to worry about it.

Reply:
All right, thanks. Do you think this process can be applied on our end so that we could build our own backport? For example. let's say this feature gets backported to `v7.x`, but we are still on `v6.y` and we can't upgrade to `v7.x` just yet. We would like to build our own patched `v6.y` version (`v6.y-patched`) to get the feature.

I can't seem to think of a simple foolproof way to backport this so that we don't have a problem when:
- upgrading from `v6.y` to `v6.y-patched`
- upgrading from `v6.y-patched` to a later version `v6.z-patched`
- upgrading from `v6-z.patched` to a vanilla `v7.x`

Maybe this is not (easily) possible, but I prefer to ask, maybe you have some idea or pointers!

Reply:
I would strongly caution you against backporting this to 6.x.  There's no clear upgrade path, since 7.x will not recognize the new serialization format comming from the modified 6.x (since the version marker will be for 7.11 or something close to that) and you'll be in unsupported territory.  Also, if we need to change the API or the serialization format for this again, you could end up in a very incompatible place.
You're welcome to try if you want to, but about all I can offer you is "good luck".

Reply:
Thank you for your advice. I think we'll skip the backport to v6 and just upgrade to v7 as soon as possible :)